specification/cognitiveservices/OpenAI.Inference/models/azure.chat.extensions.tsp

```diff
@@ -0,0 +1,75 @@
+import "@typespec/versioning";
+import "./chat.completions.tsp";
+import "./azure.cognitive.search.options.tsp";
+
+namespace Azure.OpenAI;
+
+using TypeSpec.Versioning;
+
+// Explanatory notes for Azure OpenAI chat extensions (as of 2023-08-01-preview):
+//  - Although the REST API serves this capability via a distinct /extensions/chat/completions route, client libraries
+//    should abstract that detail and dynamically dispatch this on the normal chat completions path, using the
+//    specification of extensions in the options (request body payload) as the decision mechanism
+//  - A future service API version will merge /extensions/chat/completions into /chat/completions and, at that time,
+//    the above ensures there's no customer-facing change or disruption
+//  - The current schema for specifying extensions uses a supplementary `dataSources` array at the top level of the
+//    request payload. This can be problematic for dynamic selection in some languages (minimally, ambiguous or
+//    confusing) and so libraries should instead wrap this in a custom "sub-options" type: wire format will realize as
+//    ChatCompletionsOptions.DataSources, but libraries should present that instead as
+//    ChatCompletionsOptions.AzureExtensionsOptions.Extensions
+//  - Another future service API version (TBD, perhaps the same that merges into /chat/completions) should reflect the
+//    structural and naming changes in the wire format
+
+@added(ServiceApiVersions.v2023_08_01_Preview)
+@doc("""
+  A representation of configuration data for a single Azure OpenAI chat extension. This will be used by a chat
+  completions request that should use Azure OpenAI chat extensions to augment the response behavior.
+  The use of this configuration is compatible only with Azure OpenAI.
+""")
+enum AzureChatExtensionType {
+  @doc("Represents the use of Azure Cognitive Search as an Azure OpenAI chat extension.")
+  azureCognitiveSearch: "AzureCognitiveSearch",
```

Comment:
Where does this type come from? I tried a string search in the checked out branch but couldn't find it.